app/domain/rotation/master_rotator.rb

```diff
@@ -0,0 +1,93 @@
+module Rotation
+  class MasterRotator
+
+    RotatorNotFound = ::Util::ErrorClass.new(
+      "'{0}' is not an installed rotator"
+    )
+
+    # We inject both the rotation_model and the secret_model to ease unit
+    # testing, and also because, even though the rotation query code was thrown
+    # into Secret, it should probably be refactored out into its own model
+    # class.  When that's done, it will require only a one word change here.
+    #
+    def initialize(rotators:,
+                   account:,
+                   rotation_model: Secret,
+                   secret_model: Secret)
+      @rotators = rotators
+      @account = account
+      @rotation_model = rotation_model
+      @secret_model = secret_model
+    end
+
+    def rotate_every(seconds)
+      loop do
+        rotate_all
+        sleep(seconds)
+      end
+    end
+
+    def rotate_all
+      rotations = rotation_model.required_rotations
+      Parallel.each(rotations, in_threads: 10) do |secret|
```

Comment:
What will happen if one of the threads fails?

Reply:
I'll check, but the more operative question is: What _should_ happen?  It's actually unclear to me.  We could either have a retry strategy or just fail fast.  Retry discussion aside, I _think_ that once a rotation truly fails, that's an irrecoverable exception and all of Conjur stops.  Lmk if this is not correct.  @kgilpin you thoughts on this too are appreciated.

Reply:
Use a thread pool, which will automatically re create the lost threads?